### ISSUE TITLE
Missing parts

### ISSUE THREAD

**Issue by techmach** [Creality3DPrinting/Ender-3#7]
You've left out the 250 mm long section of 4040 extrusion that serves as the center crutch of the printer.

For God's sake please Zip all the files so they can be downloaded as one download and correct the handling of STEP files.  STEP files are not to be handled as text files.

**Comment by pyromanci** [comment]
why should they zip it?
![Imgur](https://i.imgur.com/GjC62YW.png)

Also It's GitHub that is classifying the files that way. It's the site's limitations not them. Welcome to repo system. This is not Thingiverse so it has no ability to view models.

**Comment by techmach** [comment]
It's cumbersome to handle STEP files as text as well a having to deal with
them on a one on one basis.  Why not zip them a one file?

**Comment by pyromanci** [comment]
Because this is a git system. If you just have a zip file with all the step files. You change 1 file you have to rezip and re upload all the files instead of just the signal file. Which is one of the main points to a repository system such as git, cvs, svn, mercurial, ect.

Again you can just run a git clone or click the download zip button on the main page if you don't want to manually download each file. .

**Comment by techmach** [comment]
It's not an efficient use of a filing system and it still doesn't alleviate missing files.

**Comment by pyromanci** [comment]
neither does just a zip file.

**Comment by bergstar** [comment]
techmach this is not how github works. It is not a download page. Please take a moment to learn and understand git. What is "cumbersome" for you, is actually the correct way of doing things.

**Comment by techmach** [comment]
Correct is always relative to a certain POV.

**Comment by pyromanci** [comment]
@techmach are very much correct. Correct is relative to point of view. In this case Point of View is from the design intent and implementation of a repository system such a GIT.  

Which is to track and manage individual file changes. With the ability for other to fork off the files, create branches, and submit user changes/improvements to the master branch or use it as a base for their own work (like what they did with the marlin code). 

So your demand for zip files (which again, you can generate on demand right of the main page of the repository if you look) in incorrect.

Give your account creation date being the same day this repo opened. I'm guess (just like @bergstar ) this is your first time ever using a repository system and like @bergstar highly recommend you read up on what they are and how they operate.

Here are a few links to get you started:
http://lmgtfy.com/?q=how+does+github+work
http://lmgtfy.com/?q=what+is+a+code+repository

**Comment by techmach** [comment]
I've been a member under several different long forgotten usernames in the past so I know how it works.  STEP files aren't Arduino snippets nor are they TEXT files.  This system is handling them like TEXT files; they are not.   Why should I have to download a Zip of the entire upload just to get the files I need in a folder?

**Comment by techmach** [comment]
BTW, STEP files aren't user modified or edited at the character level so it's not efficient to handle them like Arduino script.  STEP files are generated by translators in the CAD systems themselves.

**Comment by pyromanci** [comment]
This site see's the step files contain only ANSI characters. So of course it's gonna register it as a text file. Complaints about it here will not do you any good. you need to go to the bottom of the page and click that Contact Github link and complain to them. As they are the only ones who can do anything about it.

However again. you replace any file. Code, image, text file, solidworks file, step file. When you replace the and commit it to the repository the older version will still be present for us to retrieve. So hand editing doesn't matter. the revision that the file changed will still be present.

Still don't understand what is so hard to just download the entire thing and delete the extra. Which i'm pretty sure is common practice for any site or download system. Thingiverse when you download a thing, you get a bunch of extra stuff with it. HP or Dell, you download drivers, which are normally a bunch of different drivers all bundled together for the same product  family. Download and SDK, you get tons of unwanted platform and examples. 

Bundling everything together for something like this IS A GOOD THING.

**Comment by techmach** [comment]
Have you been on Thingiverse lately?  You can download individual files zips, STEP's native CAD formats or STL/OBJ files independently and it seems to understand what the individual files are and how to handle them unlike NotepadHub.  I can't believe Microsoft is buying this site.

**Comment by pyromanci** [comment]
Maybe cause Billion dollar companies use this system as well. 

Of course thingiverse knows how to handle those files with viewers and custom Mime types. It's a site specific designed for 3d files. Where this system as a generic file repository.

But i will stop here. I don't think i have the brain cells left to continue this round about and how wrong you are.

**Comment by techmach** [comment]
They have viewers for STL files but that's it.  Everything else they seem to know how to handle.  It's real simple; GitHub only handles text files.  I get it.  It's not rocket science but here's the rub; Creality made a big deal out of going open source and in reality the ONLY thing they should have posted here was the Marlin files because that's the ONLY type of file this system seems to understand how to handle. 

It's your prerogative to claim I'm wrong and that may be from your POV but not being able to handle a very simple file type like a STEP or IGES file and recognizing it as a simple TEXT file is lame.  Billion dollar companies may use this site but Billion dollar companies make toys too.   For the rest of that inhabit the custom metalworking business like myself we expect a different level of expertise which is not here.  Microsoft just gave up too soon.  I bet the Empire knows how to handle STEP files.

**Comment by francisalbar** [comment]
Git repositories are not limited to text files - they work with any files including binaries, images, etc. What you are viewing right now is the Github web interface which attempts to detect mime-types and utilizes an HTML based previewer.

The web interface is not the only way to access these files. If you prefer more granularity or prefer to view these files with your own preferred viewers, please use a different interface such as a GUI Git client like SmartGit, Tortoise, or Github Desktop.

TLDR Learn to Git.

**Comment by techmach** [comment]
STEP files like DXF or IGES don't require viewers and they were never meant to be handled like simple ASCII text files.  These file types are generated via translators from 2D and 3D CAD/CAM systems.  STEP files are not meant to be edited by hand as are Arduino or Marlin files.   At issue here is that Github handles them as if a user was using snippets of code for programming devices.  The clue is the file extension which is everything in this case. The STEP extension has to mean that your system understands that this file is not a man editable file and doesn't need to be read unless the user wants to.  The end result should be that I can simply right click and save the link as a STEP file with the name and the extension intact as I would to with a *.sldprt file that is also hosted here.  

Do you understand this concept?

**Comment by francisalbar** [comment]
The Git Version Control Platform handles non text files perfectly fine. The Github WEB INTERFACE is incorrectly treating STEP files as text, and as such, the Github WEB INTERFACE is causing your issue.

If you would like the Github WEB INTERFACE to handle it appropriately, log a defect to Github and NOT this repository. Or better yet, submit a pull request that fixes the issue. You know, the way open source works. Creality has nothing to do with how the files are handled by the interface.

Your other option is to utilize a different interface to handle binary files directly as I've mentioned before. You won't have to worry about an HTML interface or your browser misrepresenting the files.
- https://git-scm.com/ Command Line
- SmartGit
- SourceTree

**Comment by techmach** [comment]
Francis,  This isn't my problem, it's Github's.  If y'all want to fix it that your decision.  As ubiquitous as STEP files and other 3D tranlation files are Github should want to make the required changes without me having to initiate a request for correction.  The Chinese are quick learners, they will figure this out.

**Comment by francisalbar** [comment]
Let me put it this way - you're posting a complaint to a television network because your old CRT TV can't display their HD broadcast correctly. Your options are to complain to the manufacturer of the TV, or get a different, FREE High Definition 4K TV. You're posting in the wrong place.

Please find the appropriate repository in https://github.com/github and post your issue there.

**Comment by techmach** [comment]
The STEP files are here in this repository Francis, not on the main page.  I 'm done.   Y'all want it your way and that's fine, you can have it your way but the problem still remains, unsolved until someone like yourself wants to make the correction.  I've spent way more time on this than it deserves.

**Comment by Autotonic** [comment]
```
STEP-File is the most widely used[citation needed] data exchange form of STEP. ISO 10303 can represent 3D objects in Computer-aided design (CAD) and related information. Due to its ASCII structure, a STEP-file is easy to read, with typically one instance per line. 
```
[Source: Wikipedia](https://en.wikipedia.org/wiki/ISO_10303)

**Comment by techmach** [comment]
Good job, Google is a great thing isn't it?  Here is some history to add to the discussion.  I was part of the STEP process back before 1994 along with thousands of my close personal friends in the metalworking business.  https://www.mmsonline.com/articles/step-ncthe-end-of-g-codes

**Comment by francisalbar** [comment]
@techmach as I've mentioned before, Creality3D, @sexycyborg, nor any of the authors for this repository are capable of fulfilling your request. Git is a platform. GitHub is just one of many hosting solutions for Git repositories. GitHub also happens to provide a closed source web interface to repositories they host, however, most users do not use this web interface to access the files. If you want them to update their web interface, please email your request support@github.com

And again, you can just as easily access this repository via other interfaces such as command line or any of the multitude of gui git tools. This will solve your issue. And since you are so fond of documentation, I highly recommend going over https://git-scm.com/book/en/v1/Getting-Started

**Comment by techmach** [comment]
STEP files are ASCII text but they are not edited by the user like an Arduino file or Marlin file.  They are translated by a CAD translator.  The SW files are corrupt and useless to anyone, even a SW user.   Your example of a model with 1000 holes is a non sequitur since the user will not go into the STEP file and change a value and save the file.  That change will come from a Translator after the model is changed so your premise is very weak.  Try again dude.  Your arguments aren't making it.  

The Contributing part is fine for Marlin code but contributing to a production solid model isn't going to work either because let's say that I import the SW files when they get fixed at some point and say that I use Pro E or NX to make changes which is highly unlikely since their face editing tools are minimal to none so those users are left to basically assembling the model in an assembly mode but they can't actually change the BREP's.  There are no translators to translate those native files back to SW so the next file type that I might use is a STEP, IGES, Parasolid or ACIS file.  How is the STEP file going to be generated?  By a translator, not by hand and not at the text level.  

I don't know what y'all do for a living but designing or machining isn't it.  Unless your CAD system is capable of hybrid modeling and face editing like ZW or IronCAD editing at the face level is not a strong suite of most proprietary CAD systems like Pro/E, NX or SW, those systems don't like to work with imported geometry.   The majority of users that are interested in a solid model like the Ender 3 aren't CAD professionals with current software that they paid for and up to date, those folks are using free stuff like F360, Freecad, Sketchup or Openscad.  

The SW models that Creality provided came from a  2016 version of SW which is now 3 years behind.  SW translators are notorious for how bad they are with STEP translations so keeping the translators up to date is paramount in the manufacturing world.   Why would you want to access STEP files at the character level and seriously mess things up?

**Comment by shoreliner** [comment]
Perhaps you should contact Github on their policy for handling these types
of files and proprietary grouping.. .I have a feeling they would be totally
confused.  I've had a full model of the printer since maybe the day after
the cad files were released.... it's aluminum extrusion, is it really that
difficult to generate for yourself for the time being?  I think some
expectations on what open hardware is and what it isn't need to be set....
this is something I will bring up during upcoming AMF meetings that involve
open hardware... I'm not sure if you understand the way Github or similar
repos/hubs work or that this isn't Customer Service. This stuff is provided
AS IS my friend.

**Comment by shoreliner** [comment]
Open Hardware is such a new concept; we've only seen it this far in things
like microcontrollers.   Are you truly having a fit over one one of the
first attempts at doing this having some issues?  This is blowing my mind.
I just want you this techmach:  I don't think this community is for you.
Sometimes we cobble things together, sometimes they don't work, sometimes
they fail.. and well, I think you might be too much of a snowflake for that
sort of an environment.  I am truly being honest, you seem to have a very
very "corporate" approach to this, indicated by your constant spouting off
of random pieces of CAD software and acronyms most of the people here don't
know and won't bother to look up.  3d printers are INCREDIBLY simple
machines, and anyone that has been in the 3d printing realm for a while
would choke to anyone this caught up over file structure. lol.

**Comment by techmach** [comment]
I'm not having a fit.  This is a discussion.  Y'all are missing the point.  I've got the bloody model.  Like shoreline said it's aluminum extrusions.  It took all of 2 hours to put it together in CAD and I did it while I was talking on the phone with an associate.   The issue is file handling and it's real simple; I'll lay it out for everyone.  STEP files are not to be handled like they are ASCII text files.  They don't work like Marlin script files because they aren't generated like Marlin files.   

File structure is important especially to Open Source.  I've been playing with these little Stepper motor incarnations for a while now and with CNC's for a lot longer.  Let's say that someone wants to design a new upgrade for their 3D printer.  They download just the affected STEP files they want, import them into their CAD system of choice and voila, they assemble just enough of what they need to prove out the design, create the model, slice it and print it; they have exact data to model from as opposed to getting out a pair of Calipers and scabbing dimensions off of an actual printer.  Isn't that what Open Source allows the owner to do?   In order for Open Source to work you have to have accurate data and the files need to be handled properly.  Just imagine if someone when into a STEP file thinking it worked like a Marlin file changed a bunch of coordinates and the uploaded it for review.  What a time waster.  Now you have a translation file that has no original model to reference.  

Don't worry though, the corporate world is coming as soon as the Empire buys this place.  If Linkedin is any indication of what is to come be prepared.  At least the Empire understands how file extensions work and what they mean.

**Comment by techmach** [comment]
BTW, STEP files are corporate files just like SW *.sldprt and *.sldasm files.  Marlin files are not.

**Comment by shoreliner** [comment]
So now you're warning of the impending takeover of Github by 'The Empire'
-- in case you weren't aware, GitHub is owned by Microsoft.  Open Source
allows YOU, or me, or anyone else to branch/merge and pull/commit these
changes, without requiring the original designer to retain the full
responsibility... this many times results in a number of branches being
created.  Some of these branches may turn into projects all thier own.  As
I had stated previously, the schematic for the missing file was there from
the start; you could have added it and put in a pull request.... but you
haven't taken the time to even learn this environment, let alone the open
source ecosystem as a whole.

If you had, you could have branched the repo, added the missing file,
corrected the assembly, and put in a PR....

Instead, you seem to be 'warning' that soon this concept will no longer
exist?  I mean, do you feel as if somehow it would make sense to 'threaten'
people that contribute their designs? lol... what planet are you from, man?

**Comment by shoreliner** [comment]
You also seem to be under the impression that github is unable to handle
non-ascii files.. or something like that? It is really unclear.  This is
100% not true and i have absolutely no idea where you got this concept
from.... are you confusing versioning and diffing?

**Comment by francisalbar** [comment]
The partial file changes are what is relevant to Git, you know, the source control system these files are currently being shared in. Whether it is auto generated or written by hand, source control tracks changes as they happen, when they happen - which, to most users, is important historical reference.

As any of these files are provided "as is", your $180 investment in a 3D printer hardly qualifies the harassment you are engaging on. You either put up, or shut up.

**Comment by shoreliner** [comment]
Can you please close this or do you want me to put the model there for you, @techmach ?

**Comment by techmach** [comment]
I'm putting up.   I already have the model so save your effort for something more productive.   I didn't post the files here, that's Creality's decision but they are here nevertheless.  I don't consider a discussion as harassment but y'all are welcome to live in the "opensource ecosystem"  for as long as you want.   Schematic's relate to wiring, electrical and electronic representation while Blueprints are related to mechanical drawings.  You should learn the difference shoreline.  

<<<"You also seem to be under the impression that github is unable to handle
non-ascii files.. or something like that? It is really unclear.  This is
100% not true and i have absolutely no idea where you got this concept
from.... are you confusing versioning and diffing?">>>  I never said they can't handle non ASCII files, you just said that.    The SW files are fine they are just corrupt because Creality didn't understand the simple concept of Pack and Go.  Github is unable to handle the STEP files properly unless you Zip the whole archive and time will prove that.  Perhaps Creality will see the light and translate their model files to something more corporate to avoid all this nonsense.  

Case closed.
